lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -1733,7 +1734,7 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
     -> (UTxOIndex WalletUTxO, Wallet s, Set Tx)
     -> SelectionStrategy
     -> PartialTx era
-    -> ExceptT ErrBalanceTx m (Cardano.Tx era)
+    -> ExceptT ErrBalanceTx m (Cardano.Tx era, s)
```

Comment:
wdyt of using a more meaningful name for the change state instead of just `s`? 
Same reasoning as why `era` type parameter is not `e`.
I'd go for `change`.

Reply:
I did consider `changeState`... The problem is that the type-signature would then currently contain `Wallet changeState`, which is misleading as `Wallet` is parameterised by the [full `s`](https://github.com/input-output-hk/cardano-wallet/blob/fcffb5b7228db71fe3382d17d708777a19fb3cca/lib/wallet/src/Cardano/Wallet/Primitive/Model.hs#L193-L194) which conceptually is a superset of `changeState`.

So I'd stick with  `s` for now, and refine further as we simplify the type-signature.